HIP/hip-584.md

```diff
@@ -0,0 +1,128 @@
+---
+hip: 584
+title: EVM Module
+author:  Ivan Kavaldzhiev <ivan.kavaldzhiev@limechain.tech>, Mustafa Uzun <mustafa.uzun@limechain.tech>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Review
+created: 2022-09-27
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/586
+updated: 2022-10-04
+---
+
+## Abstract
+
+Encapsulating Hedera smart contract EVM specific logic in a library.
+It would expose interfaces that will allow different implementations.
+Mirror node would build ethereum-related REST API on top of the library. 
+Mirror nodes could use the library to simulate a transaction and the output would be used to 
+populate the return value for the REST API that is invoked. 
+Consensus nodes would also use the library and keep their existing logic by providing specific interface implementations.
```

Comment:
```suggestion
This HIP describes the interfaces and modules that will allow the EVM implementation to be separated into a Java library. This will enable a number of future features such as different EVM implementations, EVM execution APIs on the mirror node, and transaction simulation and verification off of the consensus nodes.  The consensus nodes would continue to use this library.
```

Reply:
@shemnon, @Nana-EC  I think that saying `This HIP describes the interfaces and modules` would imply further and more detailed definitions about the implementation details for the EVM module. I think we agreed on being more abstract in terms of explanations in this HIP and if needed, link a more detailed design doc with interface definitions and examples at a later point.